Cabal/Cabal.cabal

```diff
@@ -315,6 +315,7 @@ library
     Distribution.Compat.Exception
     Distribution.Compat.Graph
     Distribution.Compat.Internal.TempFile
+    Distribution.Compat.Prelude
```

Comment:
I agree on this comment. I'd rather depend on `base-compat` for `cabal-install`, which would solve most of this PR goals.